router/sources/router.move

```diff
@@ -177,25 +185,50 @@ module router::router {
                 user,
                 domain_name,
                 registration_duration_secs,
-            )
+            );
         } else {
             abort error::not_implemented(ENOT_IMPLEMENTED_IN_MODE)
-        }
-    }
+        };
 
+        // Common operations that handle modes via the router
+        if (option::is_some(&target_addr)) {
+            set_target_addr(
+                user,
+                domain_name,
+                option::none(),
+                *option::borrow(&target_addr)
+            );
+        };
+        if (option::is_some(&to_addr)) {
+            transfer_name(user, domain_name, option::none(), *option::borrow(&to_addr));
```

Comment:
do you think we should clear the primary name during transfer? my primary name is not under my account, feels a little weird? but im fine with not clearing it